cmd/cnbBuild.go

```diff
@@ -353,6 +354,13 @@ func addProjectDescriptorTelemetryData(data *cnbBuildTelemetryData, descriptor p
 }
 
 func callCnbBuild(config *cnbBuildOptions, telemetryData *telemetry.CustomData, utils cnbutils.BuildUtils, commonPipelineEnvironment *cnbBuildCommonPipelineEnvironment, httpClient piperhttp.Sender) error {
+	tempdir, err := ioutil.TempDir("", "cnbBuild-")
+	if err != nil {
+		return errors.Wrap(err, "failed to create tempdir")
+	}
+	defer os.RemoveAll(tempdir)
+	utils.AppendEnv([]string{fmt.Sprintf("TMPDIR=%s", tempdir)})
```

Comment:
this env variable should be written to the platform folder, similarly to how we treat the `buildEnvVars`. The lifecycle binary won't pass arbitrary env variables down to the buildpacks afaik

Reply:
Good point. That is the problem with doing this now and not waiting for the buildpack to be released. It can't be tested :(